src/client/models/client/HttpClient.ts

```diff
@@ -25,6 +26,21 @@ export abstract class HttpClient {
         return `${scheme}://${this.clientOptions.requestHost}`;
     }
 
+    public getHttpProxy(): AxiosProxyConfig | false {
+        if (this.clientOptions.proxy){
+            return {
+                host: this.client.proxy.host,
```

Comment:
when adding a new feature to the library, please cover it with tests. For the case of this type of library unit tests would do, within test/unit folder.